safehttp/mux.go

```diff
@@ -164,11 +182,28 @@ func (m methodHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	for _, intercep := range m.muxInterceps {
-		if res := intercep.Before(rw, ir); res.written {
+	h.ServeHTTP(w, r)
+}
+
+// handlerWithInterceptors encapsulates a handler and its corresponding
+// interceptors.
+type handlerWithInterceptors struct {
+	handler   Handler
+	interceps map[string]Interceptor
```

Comment:
Have you considered doing this instead:

1. For each interceptor, storing both a `Interceptor` and `cfg interface{}`.
2. Pass `cfg` to `Before` and all other methods, i.e. `Before(w *safehttp.ResponseWriter, r *safehttp.IncomingRequest, cfg interface{})`?

This way we:
1. Avoid the awkward "deep-copy an interceptor" required by `Apply` right now.
2. Don't need the linear search in `context.Context` that Goa does.